binding.c

```diff
@@ -880,204 +994,247 @@ udx_napi_socket_bind (napi_env env, napi_callback_info info) {
+  assert(err == 0);
 
   udx_socket_t *self;
   size_t self_len;
-  napi_get_buffer_info(env, argv[0], (void **) &self, &self_len);
+  err = js_get_typedarray_info(env, argv[0], NULL, (void **) &self, &self_len, NULL, NULL);
+  assert(err == 0);
 
   uint32_t ttl;
-  napi_get_value_uint32(env, argv[1], &ttl);
+  err = js_get_value_uint32(env, argv[1], &ttl);
+  assert(err == 0);
 
-  int err = udx_socket_set_ttl(self, ttl);
+  err = udx_socket_set_ttl(self, ttl);
   if (err < 0) {
-    napi_throw_error(env, uv_err_name(err), uv_strerror(err));
+    err = js_throw_error(env, uv_err_name(err), uv_strerror(err));
+    assert(err == 0);
     return NULL;
   }
 
   return NULL;
 }
 
-napi_value
-udx_napi_socket_set_membership (napi_env env, napi_callback_info info) {
-  napi_value argv[4];
+js_value_t *
+udx_napi_socket_set_membership (js_env_t *env, js_callback_info_t *info) {
+  int err;
+  js_value_t *argv[4];
   size_t argc = 4;
 
-  napi_get_cb_info(env, info, &argc, argv, NULL, NULL);
+  err = js_get_callback_info(env, info, &argc, argv, NULL, NULL);
+  assert(err == 0);
 
   udx_socket_t *socket;
   size_t socket_len;
-  napi_get_buffer_info(env, argv[0], (void **) &socket, &socket_len);
+  err = js_get_typedarray_info(env, argv[0], NULL, (void **) &socket, &socket_len, NULL, NULL);
+  assert(err == 0);
 
   char mcast_addr[INET6_ADDRSTRLEN];
   size_t mcast_addr_len;
-  napi_get_value_string_utf8(env, argv[1], mcast_addr, INET6_ADDRSTRLEN, &mcast_addr_len);
+  err = js_get_value_string_utf8(env, argv[1], (utf8_t *) mcast_addr, INET6_ADDRSTRLEN, &mcast_addr_len);
+  assert(err == 0);
 
   char iface_addr[INET6_ADDRSTRLEN];
   size_t iface_addr_len;
-  napi_get_value_string_utf8(env, argv[2], iface_addr, INET6_ADDRSTRLEN, &iface_addr_len);
+  err = js_get_value_string_utf8(env, argv[2], (utf8_t *) iface_addr, INET6_ADDRSTRLEN, &iface_addr_len);
+  assert(err == 0);
 
   char *iface_param = iface_addr_len > 0 ? iface_addr : NULL;
 
   bool join; // true for join, false for leave
-  napi_get_value_bool(env, argv[3], &join);
+  err = js_get_value_bool(env, argv[3], &join);
+  assert(err == 0);
 
-  int err = udx_socket_set_membership(socket, mcast_addr, iface_param, join ? UV_JOIN_GROUP : UV_LEAVE_GROUP);
+  err = udx_socket_set_membership(socket, mcast_addr, iface_param, join ? UV_JOIN_GROUP : UV_LEAVE_GROUP);
   if (err < 0) {
-    napi_throw_error(env, uv_err_name(err), uv_strerror(err));
+    err = js_throw_error(env, uv_err_name(err), uv_strerror(err));
+    assert(err == 0);
     return NULL;
   }
 
   return NULL;
 }
 
-napi_value
-udx_napi_socket_get_recv_buffer_size (napi_env env, napi_callback_info info) {
-  napi_value argv[1];
+js_value_t *
+udx_napi_socket_get_recv_buffer_size (js_env_t *env, js_callback_info_t *info) {
+  int err;
+  js_value_t *argv[1];
   size_t argc = 1;
-  napi_get_cb_info(env, info, &argc, argv, NULL, NULL);
+  err = js_get_callback_info(env, info, &argc, argv, NULL, NULL);
+  assert(err == 0);
```

Comment:
There is some places where we check `argc` value  and in some places not (like here).

Reply:
Thanks for catching this! Yes there should be asserts everywhere, fixing